lib/services/common/securityServiceOAuth2.js

```diff
@@ -64,19 +66,29 @@ function auth(trust, callback) {
     };
 
     logger.debug(context, 'Authentication on the OAuth2 provider [%s] using trust [%s]', url, trust);
+    logger.debug(context, 'Using the following request:\n\n%s\n\n', JSON.stringify(options, null, 4));
 
     request(options, function handleResponse(error, response, body) {
         if (error) {
             logger.error(context, 'OAUTH2-001: Error retrieving token from OAuth2 provider: %s', error);
             callback(new errors.TokenRetrievalError(trust, error));
-        } else if (response.statusCode === 201) {
+        } else if (response.statusCode === 201 || response.statusCode === 200 /* Keyrock response */) {
             logger.debug(context, 'Authentication with trust [%s] was succesful', trust);
-
+            logger.debug(context,
+                'Received the following response from the OAuth2 provider:\n\n%s\n\n', JSON.stringify(body, null, 4));
             callback(null, response);
         } else if (response.statusCode === 401 || (response.statusCode === 400 &&
                 JSON.parse(response.body).error_description === 'Invalid refresh token')) {
             logger.error(context, 'Authentication rejected: %s', trust);
             callback(new errors.AuthenticationError(trust));
+        } else if (response.statusCode === 500 &&
```

Comment:
eh... this is not what the standard expect... i wonder if we should (for matter of improving the implementation of the standard) inform keyrock team about that.

Reply:
https://tools.ietf.org/html/rfc6749#section-5.2

Reply:
I will send an email to Keyrock team (and open an issue in Keyrock GE). While they fix this issue, what do you think that we must do? Wait to the fix or go on with the PR?

Reply:
I think we don't need to wait their fix, we will eventually retro fit. so this is should not block the submission, while i think the support for the 'offline_token' (i.e. never expiring refresh token) should be added, it's not much work, 1st week of october i can fix that (if you don't want to integrated it here).

Reply:
Fixed with c7d9daed5a1ebcef4a33a6c3b07809a2b8cb9588